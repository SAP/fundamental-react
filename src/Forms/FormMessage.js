import classnames from 'classnames';
import { FORM_MESSAGE_TYPES } from '../utils/constants';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/WithStyles/WithStyles';

const FormMessage = React.forwardRef(({ type, children, className, disableStyles, ...props }, ref) => {
    const formMessageClasses = classnames(
        'fd-form-message',
        {
            [`fd-form-message--${type}`]: !!type
        },
        className
    );
    return (
        <span
            {...props}
            className={formMessageClasses}
            ref={ref}>
            {children}
        </span>
    );
});

FormMessage.displayName = 'FormMessage';

FormMessage.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    customStyles: PropTypes.object,
    disableStyles: PropTypes.bool,
    type: PropTypes.oneOf(FORM_MESSAGE_TYPES)
};

export { FormMessage as __FormMessage };

export default withStyles(FormMessage, { cssFile: 'form-message', fonts: true, icons: true });
