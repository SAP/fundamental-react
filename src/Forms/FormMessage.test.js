import FormMessage from './_FormMessage';
import { mount } from 'enzyme';
import React from 'react';

describe('<FormMessage />', () => {
    describe('Prop spreading', () => {
        test('should allow props to be spread to the FormMessage component', () => {
            const element = mount(<FormMessage data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });

    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <FormMessage ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('DIV');
        expect(ref.current.className).toEqual('fd-form-message');
    });
});
