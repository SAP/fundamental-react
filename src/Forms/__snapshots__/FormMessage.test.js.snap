// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormMessage /> create form message 1`] = `
<span
  className="fd-form-message fd-form-message--help"
>
  Pellentesque metus lacus commodo eget justo ut rutrum varius nunc
</span>
`;
